api/repositories/route_repository_test.go

```diff
@@ -694,6 +708,84 @@ var _ = Describe("RouteRepository", func() {
 		})
 	})
 
+	Describe("DeleteUnmappedRoutes", func() {
+		var (
+			cfRoute1  *korifiv1alpha1.CFRoute
+			cfRoute2  *korifiv1alpha1.CFRoute
+			deleteErr error
+		)
+
+		BeforeEach(func() {
+			createRoleBinding(ctx, userName, spaceDeveloperRole.Name, space.Name)
+
+			cfRoute1 = &korifiv1alpha1.CFRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      uuid.NewString(),
+					Namespace: space.Name,
+					Labels: map[string]string{
+						korifiv1alpha1.SpaceGUIDKey: space.Name,
+					},
+				},
+				Spec: korifiv1alpha1.CFRouteSpec{
+					Host:     "my-subdomain-1",
+					Path:     "",
+					Protocol: "http",
+					DomainRef: corev1.ObjectReference{
+						Name:      domainGUID,
+						Namespace: rootNamespace,
+					},
+					Destinations: []korifiv1alpha1.Destination{
+						{
+							GUID: "destination-guid",
+							Port: tools.PtrTo[int32](8080),
+							AppRef: corev1.LocalObjectReference{
+								Name: "some-app-guid",
+							},
+							ProcessType: "web",
+							Protocol:    tools.PtrTo("http1"),
+						},
+					},
+				},
+			}
+			Expect(
+				k8sClient.Create(ctx, cfRoute1),
+			).To(Succeed())
+
+			cfRoute2 = &korifiv1alpha1.CFRoute{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      uuid.NewString(),
+					Namespace: space.Name,
+					Labels: map[string]string{
+						korifiv1alpha1.SpaceGUIDKey: space.Name,
+					},
+				},
+				Spec: korifiv1alpha1.CFRouteSpec{
+					Host:     "my-subdomain-1",
+					Path:     "",
+					Protocol: "http",
+					DomainRef: corev1.ObjectReference{
+						Name:      domainGUID,
+						Namespace: rootNamespace,
+					},
+					Destinations: []korifiv1alpha1.Destination{},
+				},
+			}
+			Expect(
+				k8sClient.Create(ctx, cfRoute2),
+			).To(Succeed())
+		})
+
+		JustBeforeEach(func() {
+			deleteErr = routeRepo.DeleteUnmappedRoutes(ctx, authInfo, space.Name)
+		})
+
+		It("deletes only the unmapped route resource", func() {
+			Expect(deleteErr).NotTo(HaveOccurred())
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: space.Name, Name: cfRoute2.Name}, &korifiv1alpha1.CFRoute{})).To(MatchError(ContainSubstring("not found")))
```

Comment:
I would suggest listing all `CFRoutes` in the `space.Name` and verify that the result only consist of the `cfRoute1` route instead of performing two gets